### ISSUE TITLE
optimization functions need an elegant copy method for creating the returned model

### ISSUE THREAD

**Issue by bwpriest** [LLNL/MuyGPyS#105]
We are currently using deepcopy, which is unnecessarily heavy (especially once we start using heteroscedastic noise). What we need is a copy method that does something like
``` 
def copy_model(model: MuyGPS) -> MuyGPS:
    ret = MuyGPS()
    # for all fixed parameters of model, set ret's corresponding references to those objects (no copying)
    # for all free parameters of model, copy ret's corresponding members to those objects (copying)
```

**Comment by bwpriest** [comment]
I think we agreed to overwrite the model. We don't need to address this for the current release, though.

**Comment by bwpriest** [comment]
This is mostly addressed by v0.7.0, although we might improve on this later.
